apps/api/src/persistence/database/arango-connection.provider.ts

```diff
@@ -24,24 +32,145 @@ const buildFullHostURL = (
 export class ArangoConnectionProvider {
     #connection: ArangoConnection;
 
-    constructor(private configService: ConfigService) {
-        const dbUser = this.configService.get<string>('ARANGO_DB_USER', 'user');
-        const dbPass = this.configService.get<string>('ARANGO_DB_USER_PASSWORD', 'userPASSWORD');
-        const dbName = this.configService.get<string>('ARANGO_DB_NAME', 'testdb');
-        const dbHostDomain = this.configService.get<string>('ARANGO_DB_HOST_DOMAIN', 'localhost');
-        const dbHostScheme = this.configService.get<Scheme>('ARANGO_DB_HOST_SCHEME', Scheme.http);
+    #databaseConfiguration: ArangoDatabaseConfiguration;
+
+    isInitialized = false;
 
-        const dbHostPort = this.configService.get<Port>('ARANGO_DB_HOST_PORT', '80');
+    public get databaseConfiguration() {
+        return this.#databaseConfiguration.getConfig();
+    }
 
-        const systemDB = new Database({
-            url: buildFullHostURL(dbHostDomain, dbHostScheme, dbHostPort),
+    constructor(private configService: ConfigService) {
+        this.setDatabaseConfiguration({
+            dbName: this.configService.get<string>('ARANGO_DB_NAME'),
+            dbRootPass: this.configService.get<string>('ARANGO_DB_ROOT_PASSWORD'),
+            dbUser: this.configService.get<string>('ARANGO_DB_USER'),
+            dbPass: this.configService.get<string>('ARANGO_DB_USER_PASSWORD'),
+            dbHostUrl: buildFullHostURL(
+                this.configService.get<string>('ARANGO_DB_HOST_DOMAIN', 'localhost'),
+                this.configService.get<Scheme>('ARANGO_DB_HOST_SCHEME', Scheme.http),
+                this.configService.get<Port>('ARANGO_DB_HOST_PORT', '80')
+            ),
         });
+
+        const systemDB = this.#getAdminDBConnection();
+
         // TODO why is `useDatabase` deprecated? can we use myDB.database("db_name")?
-        const dbInstance = systemDB.database(dbName);
-        dbInstance.useBasicAuth(dbUser, dbPass);
+        const dbInstance = systemDB.database(this.databaseConfiguration.dbName);
+        dbInstance.useBasicAuth(
+            this.databaseConfiguration.dbUser,
+            this.databaseConfiguration.dbPass
+        );
 
         this.#connection = dbInstance;
     }
 
-    getConnection = (): ArangoConnection => this.#connection;
+    /**
+     * This async setup helper will
+     * - Create the database (the one referenced in your .env via ConfigService)
+     * only if this database does not exist
+     * - Add any missing collections (but leave existing collections untouched)
+     * and can only be run once.
+     */
```

Comment:
Would we use something like this for the staging setup?

Reply:
I suppose you are getting at the fact that there is some duplication of logic with the Arango initialization script. I think the problem is the seeding of test data. You can't seed test data until the db has been initialized. But in my code path, the initialization is production code, not test code. There is 0 chance I will allow the production `Persistence Module` to put test data into the database for staging. For that reason, I think the duplication of initialization logic is not so dangerous. 

Although maybe you don't need to do any db creation or collection creation if the user says 'no' to seed test data?

Reply:
Right,so just to confirm, the e2e tests still require test data to be pre-loaded by the shell script?  And then the only other case for loading up test or dummy data by the shell script would be for the staging/demo instance?

Otherwise, it would make sense just to set up the arango instance with root password and create the coscrad user - No further action required.